### PR TITLE
Association update proposal

### PR DESCRIPTION
**Pull request by markquintontulloch**
This is what I was thinking along the lines of for Association changes

### CODE REVIEW

File: model/schema/core.yaml [hunk 1]
```diff
@@ -291,29 +291,42 @@ classes:
...
+    slots:
+      - single_reference
+
+  EvidenceAssociation:
+    abstract: true
+    is_a: Association
+    description: An association supported by any number of InformationContentEntity objects
+    slots:
+      - evidence
+
```
Comment (oblodgett) [inline model/schema/core.yaml:329]: 👍

**Review by oblodgett — APPROVED** [review]

**Comment by oblodgett** [comment]
@markquintontulloch the migration is hurting my head, but I think this is a better model.

**Comment by markquintontulloch** [comment]
> @markquintontulloch the migration is hurting my head, but I think this is a better model.

I agree that it's a better model.  I don't think the migration will be TOO bad because we only have single reference associations at the moment and all annotations will have corresponding entries in that table.  Definitely good to get this sorted before we start trying to implement any of the allele associations though.

**Review by sierra-moxon — APPROVED** [review]

**Review by chris-grove — APPROVED** [review]
Looks good. I just committed some other replacements of 'predicate' with 'relation' including several controlled vocabulary names

**Pull request merged**